src/extraction/extraction_plant.py

```diff
@@ -7,22 +7,22 @@
 # ---Imports
 import logging
 from pathlib import Path
-from typing import Any, Dict, Optional, Union
+from typing import Optional
 
 from src.beneficiations.beneficiation import Beneficiation
 from src.extraction.ingestibles import IngestibleDataclasses
+from src.extraction.metadata_extractor import IMetadataExtractor
 from src.extraction.output_manager import OutputManager
 from src.miners.miner import Miner
 from src.prospectors.prospector import Prospector
-from src.utils.types.singleton import Singleton
 
 # ---Constants
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
 
 # ---Code
-class ExtractionPlant(metaclass=Singleton):
+class ExtractionPlant(IMetadataExtractor):
     """Used for extracting data from a given profile and path. Involves prospecting, mining and beneficiating.

```

Comment:
Would be appropriate to add notes here that `ExtractionPlant` is one of two ways of creating the extractor, the other being directly implementing `IMetadataExtractor`.

Reply:
I've updated the docstring now, let me know whether it's clear enough or not